Support Alpine Linux aarch64

This is to extend the Docker set up to support M1 Mac. Currently `docker/make ready` works, but when attempting to run the tests it fails

```
$ docker/make spec.package name=Spec extra_args="--backtrace --print-perf"
On Linux, the only arch currently supported is: x86_64

On Linux, the only arch currently supported is: x86_64
rm -rf build-in-docker/llvm-static-tmp
mkdir -p build-in-docker/llvm-static-tmp
cd build-in-docker/llvm-static-tmp && curl -L --fail -sS \
        "https://github.com/savi-lang/llvm-static/releases/download/20211223/-llvm-static.tar.gz" \
| tar -xzvf -
curl: (22) The requested URL returned error: 404
gzip: invalid magic
tar: Child returned status 1
tar: Error is not recoverable: exiting now
make: *** [Makefile:182: build-in-docker/llvm-static] Error 2
```

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -36,11 +36,22 @@ task:
         # For some reason clang doesn't like it if we omit the "alpine" vendor
         # in the triple, where we'd otherwise use `x86_64-unknown-linux-musl`.
         MAKE_EXTRA_ARGS: CLANG_TARGET_PLATFORM=x86_64-alpine-linux-musl
-        # For some reason clang++ is missing some c++ headers unless we
-        # explicitly mention them in this environment variable here.
-        CPLUS_INCLUDE_PATH: /usr/include/c++/10.3.1:/usr/include/c++/10.3.1/x86_64-alpine-linux-musl
-        LIBRARY_PATH: /usr/lib/gcc/x86_64-alpine-linux-musl/10.3.1
       container:
+        image: alpine:3.15
+
+    - name: arm64-unknown-linux-musl
+      environment:
+        TRIPLE: arm64-unknown-linux-musl
+        DEPS_INSTALL: "\
+          apk add --no-cache --update \
+            bash curl \
+            alpine-sdk coreutils gcc g++ clang make linux-headers llvm12-dev \
+            pcre-dev libevent-static gc-dev crystal shards"
+        # For some reason clang doesn't like it if we omit the "alpine" vendor
+        # in the triple, where we'd otherwise use `arm64-unknown-linux-musl`.
+        # We also need to use `aarch64` instead of `arm64` here...
+        MAKE_EXTRA_ARGS: CLANG_TARGET_PLATFORM=aarch64-alpine-linux-musl
+      arm_container:
         image: alpine:3.15
 
     # # TODO: Enable FreeBSD after getting it working:

--- a/Makefile
+++ b/Makefile
@@ -124,7 +124,7 @@ CLANG_TARGET_PLATFORM?=$(TARGET_PLATFORM)
 
 # Specify where to download our pre-built LLVM/clang static libraries from.
 # This needs to get bumped explicitly here when we do a new LLVM build.
-LLVM_STATIC_RELEASE_URL?=https://github.com/savi-lang/llvm-static/releases/download/20220103
+LLVM_STATIC_RELEASE_URL?=https://github.com/savi-lang/llvm-static/releases/download/20220104
 $(eval $(call MAKE_VAR_CACHE,LLVM_STATIC_RELEASE_URL))
 
 # Specify where to download our pre-built LLVM/clang static libraries from.

--- a/platform.sh
+++ b/platform.sh
@@ -20,8 +20,14 @@ if uname | grep -iq 'Linux'; then
     else
       fail "On Linux, the supported libc variants are: gnu, musl"
     fi
+  elif uname -m | grep -iq 'aarch64'; then
+    if ldd --version 2>&1 | grep -iq musl; then
+      echo 'arm64-unknown-linux-musl'
+    else
+      fail "On arm64 Linux, the only supported libc variant is: musl"
+    fi
   else
-    fail "On Linux, the only arch currently supported is: x86_64"
+    fail "On Linux, the only arches currently supported are: x86_64, arm64"
   fi
 elif uname | grep -iq 'FreeBSD'; then
   if uname -m | grep -iq 'amd64'; then

